mcpserver/tools.go

```diff
@@ -75,30 +82,52 @@ func RunStdioServer(ctx context.Context, dag *dagger.Client) error {
 	return stdioSrv.Listen(ctx, os.Stdin, os.Stdout)
 }
 
-var tools = []*Tool{}
+var tools = []MCPHandler{}
```

Comment:
🔧 **Maintainability (High):** Having two different tool registration systems creates confusion and technical debt. Consider completing the migration or maintaining a single approach.

```suggestion
var tools = []*Tool[Request, Response]{}
```